src/Jira/Api.php

```diff
@@ -696,26 +714,32 @@ public function createRemotelink(
 		return $this->api(self::REQUEST_POST, sprintf('/rest/api/2/issue/%s/remotelink', $issue_key), $options, true);
 	}
 
-	/**
-	 * Send request to specified host.
-	 *
-	 * @param string       $method          Request method.
-	 * @param string       $url             URL.
-	 * @param array|string $data            Data.
-	 * @param boolean      $return_as_array Return results as associative array.
-	 * @param boolean      $is_file         Is file-related request.
-	 * @param boolean      $debug           Debug this request.
-	 *
-	 * @return array|Result|false
-	 */
+    /**
+     * Send request to specified host.
+     *
+     * @param string $method Request method.
+     * @param string $url URL.
+     * @param array|string $data Data.
+     * @param boolean $return_as_array Return results as associative array.
+     * @param boolean $is_file Is file-related request.
+     * @param boolean $debug Debug this request.
+     * @param array $url_params Optional parameters to add to the URL
+     *
+     * @return array|Result|false
+     */
 	public function api(
 		$method = self::REQUEST_GET,
 		$url,
 		$data = array(),
 		$return_as_array = false,
 		$is_file = false,
-		$debug = false
+		$debug = false,
+        $url_params = array()
```

Comment:
This method already has too many params, so adding another one makes it a nightmare. Any other implementations you can suggest?

My options are:

1. having special key in `$data` array that won't match any parameter of existing in JIRA API (not necessary implemented in this library) for passing query parameters
2. use `http_build_query` directly on `editIssue` method and not add any parameters here